cmd/queue/main.go

```diff
@@ -202,33 +203,33 @@ func proxyHandler(reqChan chan queue.ReqEvent, breaker *queue.Breaker, tracingEn
 	}
 }
 
-func preferPodForScaledown(logger *zap.SugaredLogger, downwardAPILabelsPath string) bool {
+func preferPodForScaledown(downwardAPILabelsPath string) (bool, error) {
 	// Short circuit a rejection when no label path file is mounted
 	if _, err := os.Stat(downwardAPILabelsPath); os.IsNotExist(err) {
-		return false
+		return false, nil
 	}
 
 	contentBytes, err := ioutil.ReadFile(downwardAPILabelsPath)
 	if err != nil {
-		logger.Warnf("Failed reading the labels file %v", err)
-		return false
+		log.Printf(err.Error())
+		return false, err
 	}
 
 	content := string(contentBytes)
 	if content == "" {
-		return false
+		return false, nil
 	}
 
 	scaleDown, err := strconv.ParseBool(content)
 	if err != nil {
-		logger.Errorf("Failed parsing the label value %v", err)
-		return false
+		log.Printf(err.Error())
```

Comment:
Let's drop the log statements and instead augment the error message with useful meta info. The caller can then log accordingly.

```suggestion
		return false, fmt.Errorf("failed parsing the label value: %w", err)
```

Same above.